portal_no_link/models/portal_no_link.py

```diff
@@ -28,6 +28,9 @@ class mail_mail(models.Model):
     @api.model
     def _get_partner_access_link(self, mail, partner=None):
         """
-        Overrides to always return nothing
+        Overrides to return nothing if partner has not user
         """
-        return None
\ No newline at end of file
+        if partner and not partner.user_ids:
```

Comment:
Shouldn't you check if user has portal group, to identify it as portal user? Portal user is also user, so this could apply to any user - portal or no portal (if you check `partner.user_ids`).